GVFS/GVFS.Mount/InProcessMount.cs

```diff
@@ -339,6 +343,28 @@ private void HandleReleaseLockRequest(string messageBody, NamedPipeServer.Connec
             connection.TrySendResponse(response.CreateMessage());
         }
 
+        private void HandlePostIndexChangedRequest(NamedPipeMessages.Message message, NamedPipeServer.Connection connection)
+        {
+            NamedPipeMessages.PostIndexChanged.Response response;
+            NamedPipeMessages.PostIndexChanged.Request request = new NamedPipeMessages.PostIndexChanged.Request(message);
+            if (request == null)
+            {
+                response = new NamedPipeMessages.PostIndexChanged.Response(NamedPipeMessages.UnknownRequest);
+            }
+            else if (this.currentState != MountState.Ready)
+            {
+                response = new NamedPipeMessages.PostIndexChanged.Response(NamedPipeMessages.MountNotReadyResult);
+            }
+            else
+            {
+                this.fileSystemCallbacks.OnIndexFileChanged(request.WorkingDirectoryUpdated, request.OnlyIndexUpdated);
+                this.fileSystemCallbacks.UpdateIndexProjection();
+                response = new NamedPipeMessages.PostIndexChanged.Response(NamedPipeMessages.PostIndexChanged.SuccessResult);

```

Comment:
Yes but the failures to update placeholders are tracked and then output at the end of the command and what would be the response if it did fail?  Throw and exception and crash?  git wouldn't care because it has already renamed the index.

Reply:
> git wouldn't care because it has already renamed the index.

If the git command running updates the index multiple times, should the command stop\fail in the middle if VFS4G fails to update its projection?

Reply:
Thinking about this and I think it would be a big change to how things have been working. Right now we only inform the user at the end of the git command of the placeholders that VFS4G was unable to update/delete.  Halting the git code in the middle could cause unknown issues especially ones like rebase that are spawning other commands like `git am`.  The outer command would need to detect the failure and do the right thing.  Git without VFS4G in many cases is a best effort anyways like when trying to delete files.  Sometimes it will inform the user and retry if the user indicates they want to but others it will just continue on.